src/Query/Terms.php

```diff
@@ -56,9 +57,15 @@ public function setTerms(array $terms): self
 
     /**
      * Adds a single term to the list.
+     *
+     * @param float|int|string $term
      */
-    public function addTerm(string $term): self
+    public function addTerm($term): self
     {
+        if (!\is_string($term) && !\is_int($term) && !\is_float($term)) {
+            throw new TypeError(\sprintf('Argument 1 passed to "%s()" must be of type float|int|string, %s given.', __METHOD__, \is_object($term) ? \get_class($term) : \gettype($term)));
```

Comment:
Can we have a `throw new \TypeError` to avoid the `use` of it at the beginning of the class?

Reply:
So what's the rule when to import and when not?

Reply:
Not sure, IMHO I would try to not import something from the root namespace, but just use `\TypeError`. WDYT?

Reply:
Yes I'm 👍 with @thePanz comment ;)

Reply:
Done!